### PR TITLE
Update renderer documentation.

### PR DESCRIPTION
**Pull request by manthey**
This is part of the gradual process of improving our documentation.

### CODE REVIEW

File: src/d3/d3Renderer.js [hunk 1]
```diff
@@ -59,6 +70,10 @@ var d3Renderer = function (arg) {
   /**
    * Meta functions for converting from geojs styles to d3.
    * @private
+   * @param {function|object} f The style value or function to convert.
+   * @param {function} [g] An optional function that returns a boolean; if it
```
Comment (aashish24) [inline src/d3/d3Renderer.js]: I am trying to remember..we are not using square braces for f because it could be an object?

Comment (manthey) [reply src/d3/d3Renderer.js]: In jsdoc brackets around a parameter indicate that it is optional.  Here, `f` is required, but `g` is optional.

File: src/d3/d3Renderer.js [hunk 2]
```diff
@@ -502,9 +569,17 @@ var d3Renderer = function (arg) {
     }
```
Comment (aashish24) [inline src/d3/d3Renderer.js]: should we return m_this here as well?

File: src/renderer.js [hunk 3]
```diff
@@ -23,28 +27,37 @@ var renderer = function (arg) {
...
+   * @param {HTMLElement} [val] If `undefined`, return the current canvas
+   *    element, otherwise set the canvas element and mark the renderer as
+   *    modified.
+   * @returns {HTMLElement|this} The current canvas element or the renderer
+   *    instance.
    */
   this.canvas = function (val) {
     if (val === undefined) {
       return m_canvas;
-    } else {
```
Comment (aashish24) [inline src/renderer.js]: +1 :)

File: src/d3/d3Renderer.js (hunk 2 above)
Comment (manthey) [reply src/d3/d3Renderer.js]: This is called by `renderAnimationFrame`, so any return value will be ignored.

File: src/d3/d3Renderer.js (hunk 1 above)
Comment (aashish24) [reply src/d3/d3Renderer.js]: ah.. that's it. thanks for reminding that.

**Pull request merged**